### PR TITLE
Deprecate `checkpoint_callback` from the `Trainer` constructor in favour of `enable_checkpointing`

### LINKED ISSUE #9665: Rename `checkpoint_callback` to `enable_checkpointing` in Trainer constructor
## Proposed refactoring or deprecation

Rename `checkpoint_callback` to `enable_checkpointing` in Trainer constructor

### Motivation

`checkpoint_callback` in the Trainer constructor serves as a flag to enable/disable checkpointing. This can be confusing to users, especially since the Trainer has a `checkpoint_callback` _property_ which is what is sounds like, an instance of ModelCheckpoint. A more appropriate arg name would be `enable_checkpointing`

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixes #9665

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by kaushikb11** [comment]
IMO, it should be `enable_checkpoint` instead

**Comment by rohitgr7** [comment]
okay.. 
small poll for anyone who sees this
enable_checkpointing -> 👍 
enable_checkpoint -> 🎉

**Comment by tchaton** [comment]
> okay.. small poll for anyone who sees this enable_checkpointing -> 👍 enable_checkpoint -> 🎉 enable_checkpoints -> 🚀

Hey @rohitgr7,

Quick question. enable_checkpointing is True by default right ? Won't it be more intuitive to have `disable_checkpointing=False` ?

Best,
T.C

**Comment by rohitgr7** [comment]
> > okay.. small poll for anyone who sees this enable_checkpointing -> 👍 enable_checkpoint -> 🎉 enable_checkpoints -> 🚀
> 
> Hey @rohitgr7,
> 
> Quick question. enable_checkpointing is True by default right ? Won't it be more intuitive to have `disable_checkpointing=False` ?
> 
> Best, T.C

@tchaton 
- this will flip the value we currently have with `checkpoint_callback`. I think just renaming is fine. so if someone is switching they just need to flip the key, that's all.
- we have other `enable_*` flags inside trainer but no `disable_*` flags.

File: tests/accelerators/test_tpu_backend.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def test_resume_training_on_cpu(tmpdir):
     """Checks if training can be resumed from a saved checkpoint on CPU."""
     # Train a model on TPU
     model = BoringModel()
-    trainer = Trainer(checkpoint_callback=True, max_epochs=1, tpu_cores=8)
+    trainer = Trainer(enable_checkpointing=True, max_epochs=1, tpu_cores=8)
```
Comment (ananthsub) [inline tests/accelerators/test_tpu_backend.py:53]: why's this needed if it's the default value? 
```suggestion
    trainer = Trainer(max_epochs=1, tpu_cores=8)
```

File: tests/accelerators/test_tpu_backend.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def test_resume_training_on_cpu(tmpdir):
 
     # Verify that training is resumed on CPU
     trainer = Trainer(
-        resume_from_checkpoint=model_path, checkpoint_callback=True, max_epochs=1, default_root_dir=tmpdir
+        resume_from_checkpoint=model_path, enable_checkpointing=True, max_epochs=1, default_root_dir=tmpdir
```
Comment (ananthsub) [inline tests/accelerators/test_tpu_backend.py:65]: same here?
```suggestion
        resume_from_checkpoint=model_path, max_epochs=1, default_root_dir=tmpdir
```

File: tests/accelerators/test_tpu_backend.py [hunk 3]
```diff
@@ -86,7 +86,7 @@ def test_weight_tying_warning(tmpdir, capsys=None):
     """Ensure a warning is thrown if model parameter lengths do not match post moving to device."""
 
     model = WeightSharingModule()
-    trainer = Trainer(checkpoint_callback=True, max_epochs=1, tpu_cores=1)
+    trainer = Trainer(enable_checkpointing=True, max_epochs=1, tpu_cores=1)
```
Comment (ananthsub) [inline tests/accelerators/test_tpu_backend.py:89]:
```suggestion
    trainer = Trainer(max_epochs=1, tpu_cores=1)
```

File: tests/accelerators/test_tpu_backend.py [hunk 4]
```diff
@@ -105,7 +105,7 @@ def on_post_move_to_device(self):
             self.layer_3.weight = self.layer_1.weight
 
     model = Model()
-    trainer = Trainer(checkpoint_callback=True, max_epochs=1, tpu_cores=1)
+    trainer = Trainer(enable_checkpointing=True, max_epochs=1, tpu_cores=1)
```
Comment (ananthsub) [inline tests/accelerators/test_tpu_backend.py:108]:
```suggestion
    trainer = Trainer(max_epochs=1, tpu_cores=1)
```

File: tests/checkpointing/test_legacy_checkpoints.py [hunk 5]
```diff
@@ -79,7 +79,7 @@ def test_resume_legacy_checkpoints(tmpdir, pl_version: str):
             default_root_dir=str(tmpdir),
             gpus=int(torch.cuda.is_available()),
             precision=(16 if torch.cuda.is_available() else 32),
-            checkpoint_callback=True,
+            enable_checkpointing=True,
```
Comment (ananthsub) [inline tests/checkpointing/test_legacy_checkpoints.py:82]:
```suggestion
```

File: tests/checkpointing/test_model_checkpoint.py [hunk 6]
```diff
@@ -1008,17 +1008,17 @@ def test_configure_model_checkpoint(tmpdir):
...
 
     # no callbacks
-    trainer = Trainer(checkpoint_callback=False, callbacks=[], **kwargs)
+    trainer = Trainer(enable_checkpointing=False, callbacks=[], **kwargs)
     assert not any(isinstance(c, ModelCheckpoint) for c in trainer.callbacks)
     assert trainer.checkpoint_callback is None
 
     # default configuration
-    trainer = Trainer(checkpoint_callback=True, callbacks=[], **kwargs)
+    trainer = Trainer(enable_checkpointing=True, callbacks=[], **kwargs)
```
Comment (ananthsub) [inline tests/checkpointing/test_model_checkpoint.py:1016]:
```suggestion
    trainer = Trainer(callbacks=[], **kwargs)
```

File: tests/loggers/test_all.py [hunk 7]
```diff
@@ -338,7 +338,7 @@ def _test_logger_created_on_rank_zero_only(tmpdir, logger_class):
         accelerator="ddp_cpu",
         num_processes=2,
         max_steps=1,
-        checkpoint_callback=True,
+        enable_checkpointing=True,
```
Comment (ananthsub) [inline tests/loggers/test_all.py:341]:
```suggestion
```

File: tests/accelerators/test_tpu_backend.py (hunk 1 above)
Comment (rohitgr7) [reply tests/accelerators/test_tpu_backend.py:53]: I just did the replacements all at once.. you don't expect me to go, search and replace in each file .. haha :sweat_smile: 
thanks for pointing out btw.. :)

File: tests/loops/test_loops.py [hunk 8]
```diff
@@ -565,7 +565,7 @@ def train_dataloader(self):
         accumulate_grad_batches=accumulate_grad_batches,
         enable_progress_bar=False,
         logger=False,
-        checkpoint_callback=True,
+        enable_checkpointing=True,
```
Comment (ananthsub) [inline tests/loops/test_loops.py:568]:
```suggestion
```

**Comment by tchaton** [comment]
> * also I think it's better to ask user, do you want checkpointing or not? rather than do you not want checkpointing or do you? :|=

Thanks @rohit. Yes would be. But at the same time, it is an opt-out feature by default, so users would do enable_checkpoint=False which doesn't convey the default behaviour at first glance.

Best,
T.C
